File: src/main/java/com/ibmf/jigarthakkar/amfi/service/UserService.java
package com.ibmf.jigarthakkar.amfi.service;

import java.util.List;

import com.ibmf.jigarthakkar.amfi.dto.UserDto;

/**
 * Created by ashish on 13/5/17.
 */
public interface UserService {
    UserDto getUserById(Integer userId);
    void saveUser(UserDto userDto);
    List<UserDto> getAllUsers();
}

File: src/main/resources/static/js/app/UserService.js
'use strict'

angular.module('demo.services', []).factory('UserService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getUserById = function(userId) {
				var url = CONSTANTS.getUserByIdUrl + userId;
				return $http.get(url);
			}
			service.getAllUsers = function() {
				return $http.get(CONSTANTS.getAllUsers);
			}
			service.saveUser = function(userDto) {
				return $http.post(CONSTANTS.saveUser, userDto);
			}
			service.getAllFunds = function() {
				return $http.get(CONSTANTS.getAllFunds);
			}

			service.getSchemeDataByFund = function(fundSchemeDto) {
				return $http.post(CONSTANTS.getSchemeByFund, fundSchemeDto);
			}

			return service;
		} ]);